From: Sven Eckelmann <sven@narfation.org>
Date: Fri, 10 Jun 2016 19:22:30 +0200
Subject: [PATCH] batman-adv: Print bcast_full errors via ratelimited function

Helps to identify problems due to bcast storms. This may be related to
missing DHCP Offer replies which never end up at the client.

Signed-off-by: Sven Eckelmann <sven@narfation.org>
---
 net/batman-adv/send.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/batman-adv/send.c b/net/batman-adv/send.c
index d7195f5..ed07adf 100644
--- a/net/batman-adv/send.c
+++ b/net/batman-adv/send.c
@@ -496,8 +496,8 @@ int batadv_add_bcast_packet_to_list(struct batadv_priv *bat_priv,
 	struct sk_buff *newskb;
 
 	if (!batadv_atomic_dec_not_zero(&bat_priv->bcast_queue_left)) {
-		batadv_dbg(BATADV_DBG_BATMAN, bat_priv,
-			   "bcast packet queue full\n");
+		net_ratelimited_function(batadv_err, bat_priv->soft_iface,
+					 "bcast packet queue full\n");
 		goto out;
 	}
 
